### PR TITLE
Check permissions when deleting comments

### PR DESCRIPTION
**Pull request by TsarN**
Sorry for a relatively large PR, I did a little bit of refactoring.

### CODE REVIEW

File: portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    datastore.put(comment);
+
+    when(request.getParameter("id")).thenReturn(KeyFactory.keyToString(comment.getKey()));
+    new CommentDeleteServlet().doPost(request, response);
+
+    verify(response).sendError(HttpServletResponse.SC_FORBIDDEN, "not allowed to delete the comment");
+
+    Query query = new Query("comment");
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+    assertThat(results.toArray(), arrayContaining(comment));
```
Comment (DarthGandalf) [inline portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java:52]: `assertThat(results, contains(comment))` should be simpler

File: portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import static org.hamcrest.Matchers.arrayContaining;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class TestCommentDeleteServlet extends ServletTest {
+  @Test
+  public void testUnauthenticatedDelete() throws Exception {
+    helper.setEnvIsLoggedIn(false);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (DarthGandalf) [inline portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java:35]: Does this access production DB, or is it a test one?

File: portfolio/src/test/java/com/google/sps/TestCommentDeleteAllServlet.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    datastore.put(comment1);
+    datastore.put(comment2);
+
+    new CommentDeleteAllServlet().doPost(request, response);
+
+    verify(response).setStatus(HttpServletResponse.SC_NO_CONTENT);
+    Query query = new Query("comment");
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+    assertThat(results.size(), equalTo(0));
```
Comment (DarthGandalf) [inline portfolio/src/test/java/com/google/sps/TestCommentDeleteAllServlet.java:41]: hasSize()? to show contents of the list if it's wrong. I even see `IsEmptyCollection` in hamcrest

File: portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    verify(response).sendError(HttpServletResponse.SC_FORBIDDEN, "not allowed to delete the comment");
+
+    Query query = new Query("comment");
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+    assertThat(results.toArray(), arrayContaining(comment));
+  }
+
+  @Test
+  public void testAdminDelete() throws Exception {
+    helper.setEnvIsAdmin(true);
```
Comment (DarthGandalf) [inline portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java:79]: Where is this function defined?

File: portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java (hunk 2 above)
Comment (TsarN) [reply portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java:35]: The class inherits from ServletTest that sets up a test DB

File: portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java (hunk 4 above)
Comment (TsarN) [reply portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java:79]: In the base class, ServletTest

Comment (DarthGandalf) [reply portfolio/src/test/java/com/google/sps/TestCommentDeleteServlet.java:78]: Ah, I thought you need to implement admin bit of user yourself

**Pull request merged**